src/main/java/uk/gov/companieshouse/company/links/processor/CompanyProfileStreamProcessor.java

```diff
@@ -1,19 +1,115 @@
 package uk.gov.companieshouse.company.links.processor;
 
+import org.apache.commons.lang3.StringUtils;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.messaging.Message;
 import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.company.CompanyProfile;
+import uk.gov.companieshouse.api.company.Data;
+import uk.gov.companieshouse.api.company.Links;
+import uk.gov.companieshouse.api.model.ApiResponse;
+import uk.gov.companieshouse.api.psc.PscList;
+import uk.gov.companieshouse.company.links.exception.NonRetryableErrorException;
+import uk.gov.companieshouse.company.links.exception.RetryableErrorException;
+import uk.gov.companieshouse.company.links.logging.DataMapHolder;
 import uk.gov.companieshouse.company.links.service.CompanyProfileService;
+import uk.gov.companieshouse.company.links.service.PscService;
+import uk.gov.companieshouse.company.links.type.ApiType;
 import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.stream.ResourceChangedData;
 
 @Component
 public class CompanyProfileStreamProcessor extends StreamResponseProcessor {
 
     private final CompanyProfileService companyProfileService;
+    private final PscService pscService;
 
+    /**
+     * Construct a Company Profile stream processor.
+     */
     @Autowired
     public CompanyProfileStreamProcessor(CompanyProfileService companyProfileService,
-                                         Logger logger) {
+                                         PscService pscService, Logger logger) {
         super(logger);
         this.companyProfileService = companyProfileService;
+        this.pscService = pscService;
+    }
+
+    /**
+     * Process a ResourceChangedData message.
+     */
+    public void processDelta(Message<ResourceChangedData> resourceChangedMessage) {
+        final ResourceChangedData payload = resourceChangedMessage.getPayload();
+        final String contextId = payload.getContextId();
+        final String companyNumber = payload.getResourceId();
+        DataMapHolder.get()
+                .companyNumber(companyNumber);
+        final ApiResponse<CompanyProfile> response = getCompanyProfile(payload, companyNumber);
+        var data = response.getData().getData();
+
+        if (data.getLinks() != null) {
```

Comment:
not sure we want this null check, if the existing links object is null we should still check to see if there are any PSCs?